From 19a1ffc1450bd3ecfb7419d74c733b2e16bf47f3 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Wed, 26 Aug 2015 08:14:41 +0200
Subject: [PATCH 11/12] Let configure pass on unknown architectures setting
 sane defaults

This helps porting Kodi to new architectures while emitting warnings about
entering uncharted territory.
---
 configure.ac    | 3 ++-
 m4/xbmc_arch.m4 | 6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 925466a..8369615 100644
--- a/configure.ac
+++ b/configure.ac
@@ -658,7 +658,8 @@ case $host in
      USE_STATIC_FFMPEG=1
      ;;
   *)
-     AC_MSG_ERROR(unsupported host ($host))
+    ARCH=$host_cpu"-"$host_os
+    AC_MSG_WARN([unsupported host ($host), guessing ARCH as $ARCH.])
 esac
 AC_SUBST([ARCH])
 
diff --git a/m4/xbmc_arch.m4 b/m4/xbmc_arch.m4
index ff5dc20..dfb45fd 100644
--- a/m4/xbmc_arch.m4
+++ b/m4/xbmc_arch.m4
@@ -27,7 +27,8 @@ case $build in
      AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
      ;;
   *)
-     AC_MSG_ERROR(unsupported native build platform: $build)
+     AC_MSG_WARN([unsupported native build platform: $build])
+     AC_SUBST(NATIVE_ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
 esac
 
 
@@ -70,7 +71,8 @@ case $host in
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -DTARGET_ANDROID")
      ;;
   *)
-     AC_MSG_ERROR(unsupported build target: $host)
+     AC_MSG_WARN([unsupported native build platform: $build])
+     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
 esac
 
 if test "$target_platform" = "target_android" ; then
-- 
2.10.2

